src/backend/distributed/utils/node_metadata.c

```diff
@@ -924,7 +962,7 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 
 	DeleteNodeRow(nodeName, nodePort);
 
-	if (WorkerNodeIsPrimary(workerNode))
+	if (WorkerNodeIsPrimaryCurrentDataNode(workerNode))
 	{
 		UpdateColocationGroupReplicationFactorForReferenceTables(
 			ActivePrimaryNodeCount());
```

Comment:
It feels like this count needs to be reflecting the change in how we decide which nodes have a reference table on them.